soweego/importer/imdb_dump_extractor.py

```diff
@@ -0,0 +1,390 @@
+                        imdb_entity.ImdbProducerEntity(),
+                        imdb_entity.ImdbWriterEntity(),
+                    ]
+
+                # add person to every matching table
+                for etype in types_of_entities:
+                    self._populate_person(etype, person_info, session)
+
+                # if person is known for any movies then add these to the
+                # databse as well
+                if person_info.get("knownForTitles"):
+                    self.n_person_movie_links += 1
+                    self._populate_person_movie_relations(person_info, session)
+
+                # commit results to the database every `_sqlalchemy_commit_every` loops
+                if self.n_persons % self._sqlalchemy_commit_every == 0:
+                    session.commit()
+
+                self.n_persons += 1
+
+            # finally commit remaining entities
+            session.commit()
+
+        end = datetime.datetime.now()
+        LOGGER.info("Person import completed in %s. "
+                    "Total people imported: %d - "
+                    "Actors: %d - Directors: %d - Musicians: %d - "
+                    "Producers: %d - Writers: %d - Misc: %d",
+                    end - start, self.n_persons, self.n_actors,
+                    self.n_directors, self.n_musicians, self.n_producers,
+                    self.n_writers, self.n_misc)
+
+    def _populate_person(self, person_entity: imdb_entity.ImdbPersonEntity,
+                         person_info: Dict,
+                         session: object) -> None:
+        """
+        Given an instance of
+        :ref:`soweego.importer.models.imdb_entity.ImdbPersonEntity`
+        this function populates its attributes according to
+        the provided `person_info` dictionary. It then adds
+        said instance to the SQLAlchemy session.
+
+        :param person_entity: the entity which we want to populate
+        :param person_info: the data we want to populate the
+        entity with
+        :param session: the SQLAlchemy session to which we will
+        add the entity once it is populated.
+        """
+
+        person_entity.catalog_id = person_info.get("nconst")
+        person_entity.name = person_info.get("primaryName")
+        person_entity.name_tokens = " ".join(
+            text_utils.tokenize(person_entity.name))
+
+        # If either `actor` or `actress` in primary profession
+        # (which is a comma separated string of professions)
+        # then we can distinguish the gender
+        if any(prof in person_info.get("primaryProfession")
+               for prof in ["actor", "actress"]):
+            person_entity.gender = "male" if "actor" in person_info.get(
+                "primaryProfession") else "female"
+
+        # datetime.date(year, month, day)
+        born_year = person_info.get("birthYear")
+        if born_year:
+            person_entity.born = datetime.date(int(born_year), 1, 1)
```

Comment:
I believe it's a strong assumption to set month and day to `1` as default values. Would it be possible to avoid this?

Reply:
My bad, I should have added a comment to clarify what is happening here. If you see the definition of `ImdbPersonEntity` you'll see that the born and died precision is _9_. 

https://github.com/tupini07/soweego/blob/3953fcddea2a0744a4f2f1331d1e0478e0ddc1e3/soweego/importer/models/imdb_entity.py#L65-L67

From what I see in `vocabulary.py`, a date precision of 9 means that only the year is correct. Let me know if I'm mistaken :smile: 

In the meantime I've added a comment to clarify this, in 73172e5d5956527ca7bb95e9a196758925f265c8